platform-common-pdf/src/test/java/com/softicar/platform/common/pdf/PdfPageRemoverTest.java

```diff
@@ -0,0 +1,37 @@
+package com.softicar.platform.common.pdf;
+
+import com.softicar.platform.common.io.buffer.ByteBuffer;
+import com.softicar.platform.common.testing.AbstractTest;
+import java.io.IOException;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.text.PDFTextStripper;
+import org.junit.Test;
+
+public class PdfPageRemoverTest extends AbstractTest {
+
+	@Test
+	public void testRemoveLeadingBlankPagesWithLeadingBlankPages() throws IOException {
+
+		byte[] inputPdfBytes = new ByteBuffer(PdfTestFiles.PDF_WITH_TWO_LEADING_BLANK_PAGES.getResource()::getResourceAsStream).getBytes();
+
+		byte[] outputPdfBytes = new PdfPageRemover().removeLeadingBlankPages(inputPdfBytes);
+
+		try (var document = PDDocument.load(outputPdfBytes)) {
+			assertEquals(1, document.getNumberOfPages());
+			assertEquals("Third page\n", new PDFTextStripper().getText(document));
+		}
+	}
+
+	@Test
+	public void testRemoveLeadingBlankPagesWithoutLeadingBlankPages() throws IOException {
```

Comment:
- the name `testRemoveLeadingBlankPagesWithoutLeadingBlankPages` is extremely confusing, you have to read the test to understand ;)
- suggest `testRemoveLeadingBlankPagesWithPdfWithNoLeadingBlankPages` or similar
- I am missing a test with multiple blank pages interleaved with non-blank pages to prevent regressions.
- also the name `PDF_WITHOUT_LEADING_BLANK_PAGES` does not indicate if there are any non-leading blank pages, is that intentional?

Reply:
All of this was reworked to properly reflect that intermediate blank pages are removed as well. Pls take a look.